src/main/java/fr/skytasul/guardianbeam/Laser.java

```diff
@@ -391,15 +391,23 @@ protected void sendDestroyPackets(Player p) throws ReflectiveOperationException
 		@Override
 		public void moveStart(Location location) throws ReflectiveOperationException {
 			this.start = location;
-			if (main != null) moveFakeEntity(start, guardianID, guardian);
-		}
+			if (main != null) {
+        		initGuardian();
+        		if (guardian == null) {
+            		for (Player p : show) {
+            		   Packets.sendPackets(p, createGuardianPacket, metadataPacketGuardian);
+            		}
```

Comment:
Guardian should never be null as it is initialized at laser start and will never be null again.